storage-quickstart.go

```diff
@@ -26,107 +24,88 @@ import (
+	url := "https://<storage-account-name>.blob.core.windows.net/"
 	ctx := context.Background()
 
-	// Create a default request pipeline using your storage account name and account key.
 	credential, err := azidentity.NewDefaultAzureCredential(nil)
-	if err != nil {
-		log.Fatal("Invalid credentials with error: " + err.Error())
-	}
+	handleError(err)
 
-	serviceClient, err := azblob.NewServiceClient(url, credential, nil)
-	if err != nil {
-		log.Fatal("Invalid credentials with error: " + err.Error())
-	}
+	client, err := azblob.NewClient(url, credential, nil)
+	handleError(err)
+
+	serviceClient := client.ServiceClient()
 
 	// Create the container
-	containerName := fmt.Sprintf("quickstart-%s", randomString())
+	containerName := "quickstart-sample-container"
 	fmt.Printf("Creating a container named %s\n", containerName)
 	containerClient := serviceClient.NewContainerClient(containerName)
 	_, err = containerClient.Create(ctx, nil)
-	if err != nil {
-		log.Fatal(err)
-	}
-
-	fmt.Printf("Creating a dummy file to test the upload and download\n")
+	handleError(err)
 
-	data := []byte("\nhello world this is a blob\n")
-	blobName := "quickstartblob" + "-" + randomString()
+	data := []byte("\nHello, world! This is a blob.\n")
+	blobName := "sample-blob"
 
-	blobClient, err := azblob.NewBlockBlobClient(url+containerName+"/"+blobName, credential, nil)
-	if err != nil {
-		log.Fatal(err)
-	}
+	blobClient := containerClient.NewBlockBlobClient(blobName)
 
 	// Upload to data to blob storage
-	_, err = blobClient.UploadBufferToBlockBlob(ctx, data, azblob.HighLevelUploadToBlockBlobOption{})
-
-	if err != nil {
-		log.Fatalf("Failure to upload to blob: %+v", err)
-	}
+	fmt.Printf("Uploading a blob named %s\n", blobName)
+	_, err = blobClient.UploadBuffer(ctx, data, &azblob.UploadBufferOptions{})
+	handleError(err)
 
 	// List the blobs in the container
 	fmt.Println("Listing the blobs in the container:")
 
-	pager := containerClient.ListBlobsFlat(nil)
+	pager := containerClient.NewListBlobsFlatPager(&container.ListBlobsFlatOptions{
+		Include: container.ListBlobsInclude{Snapshots: true, Versions: true},
+	})
 
-	for pager.NextPage(ctx) {
-		resp := pager.PageResponse()
+	for pager.More() {
+		resp, err := pager.NextPage(context.TODO())
+		handleError(err)
 
-		for _, v := range resp.ContainerListBlobFlatSegmentResult.Segment.BlobItems {
-			fmt.Println(*v.Name)
+		for _, blob := range resp.Segment.BlobItems {
+			fmt.Println(*blob.Name)
 		}
 	}
 
-	if err = pager.Err(); err != nil {
-		log.Fatalf("Failure to list blobs: %+v", err)
-	}
-
 	// Download the blob
-	get, err := blobClient.Download(ctx, nil)
-	if err != nil {
-		log.Fatal(err)
-	}
+	get, err := blobClient.DownloadStream(ctx, nil)
+	handleError(err)
 
-	downloadedData := &bytes.Buffer{}
-	reader := get.Body(azblob.RetryReaderOptions{})
+	downloadedData := bytes.Buffer{}
+	reader := get.Body
 	_, err = downloadedData.ReadFrom(reader)
```

Comment:
This is the simple case but has reliability problems (i.e. no retries on an interrupted stream).  Consider switching this out to something like this.
```go
rr := get.NewRetryReader(...)
downloadedData.ReadFrom(rr)
```